python/lsst/sims/catalogs/definitions/ParallelCatalogWriter.py

```diff
@@ -0,0 +1,113 @@
+import copy
+import time
+
+
+__all__ = ["parallelCatalogWriter"]
+
+
+def parallelCatalogWriter(catalog_dict, chunk_size=None, constraint=None,
+                          write_mode='w', write_header=True):
+    """
+    This method will take several InstanceCatalog classes that are meant
+    to be based on the same CatalogDBObject and write them out in parallel
+    from a single database query.  The imagined use-case is simultaneously
+    writing out a PhoSim InstanceCatalog as well as the truth catalog with
+    the pre-calculated positions and magnitudes of the sources.
+
+    Parameters
+    ----------
+    catalog_dict is a dict keyed on the names of the files to be written.
+    The values are the IntanceCatalogs to be written (note: these are full
+    instantiations of InstanceCatalogs, not just InstanceCatalog classes
+    as with the CompoundInstanceCatalog)
+
+    constraint is an optional SQL constraint to be applied to the database query.
+    Note: constraints applied to individual catalogs will be ignored.
+
+    chunk_size is an int which optionally specifies the number of rows to be
+    returned from db_obj at a time
+
+    write_mode is either 'w' (write) or 'a' (append), determining whether or
+    not the writer will overwrite existing catalog files (assuming they exist)
+
+    write_header is a boolean that controls whether or not to write the header
+    in the catalogs.
+
+    Output
+    ------
+    This method does not return anything, it just writes the files that are the
+    keys of catalog_dict
+    """
+
+    list_of_file_names = catalog_dict.keys()
+    ref_cat = catalog_dict[list_of_file_names[0]]
+    for ix, file_name in enumerate(list_of_file_names):
+        if ix>0:
+            cat = catalog_dict[file_name]
+            try:
+                assert cat.obs_metadata == ref_cat.obs_metadata
+            except:
+                print cat.obs_metadata
+                print ref_cat.obs_metadata
+                raise RuntimeError('Catalogs passed to parallelCatalogWriter have different '
+                                   'ObservationMetaData.  I do not know how to deal with that.')
+
+            try:
```

Comment:
Mostly because `DBConnection.__eq__` also tests on `verbose`.  I removed `verbose` from `DBConnection.__eq__` and used `==` here.